src/GitHub.Services.Vssdk/Services/TippingService.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Windows;
+using GitHub.Logging;
+using Microsoft;
+using Serilog;
+using IServiceProvider = System.IServiceProvider;
+
+namespace GitHub.Services.Vssdk.Services
+{
+    /// <summary>
+    /// This service is a thin wrapper around <see cref="Microsoft.Internal.VisualStudio.Shell.Interop.IVsTippingService"/>.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="IVsTippingService"/> interface is public, but contained within the 'Microsoft.VisualStudio.Shell.UI.Internal' assembly.
+    /// To avoid a direct dependency on 'Microsoft.VisualStudio.Shell.UI.Internal', we use reflection to call this service.
```

Comment:
I'm curious about this. Since you have an *effective* dependency on `Microsoft.VisualStudio.Shell.UI.Internal`, what is the downside to expressing that via direct dependency?

Reply:
Ah, I think I answered my own question by reading below:

> // The option to pass a command option is only available on Visual Studio 2017+.